addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -94,18 +94,88 @@ function factory(dependencies) {
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         * @private
+         * @returns {mail.message[]}
+         */
+        _computeFetchedMessages() {
+            if (!this.thread) {
+                return [];
```

Comment:
Would be safer to unlink-all? I don't think we want to keep previous value if thread is hypothetically removed. (or thread is made readonly and required and then condition does not make sense).

Reply:
thread -> cache is causal, so if thread is deleted cache is deleted too, but ok for `unlink-all`